Utilities/bootstrap

```diff
@@ -805,7 +805,7 @@ def get_swiftpm_flags(args):
 
     build_target = get_build_target(args)
     cross_compile_hosts = args.cross_compile_hosts
-    if build_target == 'x86_64-apple-macosx' and "macosx-arm64" in cross_compile_hosts:
+    if (build_target == 'x86_64-apple-macosx' or build_target == 'arm64-apple-macosx') and "macosx-arm64" in cross_compile_hosts:
```

Comment:
This doesn't do what you think it does: it will have no effect on building SPM for an M1 mac.

What this does is that if the host is an M1 mac and you somehow pass in macosx-arm64 as a cross-compilation host, it will build SPM for x86_64 too, which makes no sense.

Reply:
Not sure of the specifics myself but this change made the difference between being able to build a toolchain using swift/utils/build-toolchain and receiving the error a couple of lines down.

Reply:
It's quite unfortunate that the cross-compilation here is so asymmetrical.  As @johnno1962 points out it isn't currently possible to build a universal toolchain on M1 because of the existing check for the host architecture, and this change does seem to fix that, though the logic looks as if it doesn't make any sense.  I think this is because the logic that builds the toolchain passes `-cross-compile-hosts macos-arm64` to mean "build a universal toolchain for x86_64 and arm64", which is what doesn't make sense to begin with.

@shahmishal if this were cleaned up to essentially say `--build-universal` and imply the same semantics, could the scripts that invoke this logic as part of the toolchain build be fixed?  It would be good to clean this up now that more people are using M1 Macs.

Reply:
OK, I see what you mean now: the issue is that [the base mac preset assumes that it is building on an Intel Mac and passes arm64 as a cross-compilation host](https://github.com/apple/swift/blob/afb3110e21c1dbc36287e96c5187200d5f0d98c5/utils/build-presets.ini#L1200). This change is merely a hack to make that work, you should be fixing those presets instead.

Reply:
Thanks for picking this up.  I've updated the ticket documenting my attempts to build a toolchain and the various "hacks" I've had to apply here: https://bugs.swift.org/browse/SR-14035.

Reply:
> OK, I see what you mean now: the issue is that [the base mac preset assumes that it is building on an Intel Mac and passes arm64 as a cross-compilation host](https://github.com/apple/swift/blob/afb3110e21c1dbc36287e96c5187200d5f0d98c5/utils/build-presets.ini#L1200). This change is merely a hack to make that work, you should be fixing those presets instead.

It seems to me that part of the problem here is the naming of the bootstrap option, and the fact that it depends on the host architecture.  So I think a good way forward here would be to make this PR change the option to more clearly specific the intent of building for all appropriate architectures (for any platform, in theory, but currently specifically for arm64 and x86_64 on macOS), and then to change the presets to use that flag.

We can't just remove the existing flag, so probably something like:
- add a new "-build-for-all-architectures" or "-build-universal" or something like that, which semantics that don't depend on the host but that just on macOS add both architectures, like the current logic
- modify the presets to use that flag
- remove or deprecate the current flag

Reply:
It does not seem as if there are other settings in that preset that tie the building of the toolchain to x86_64 hosts.

Reply:
Adding a universal flag would work and yes, this is the only preset flag that implies building on an Intel mac, but I think it was a mistake to build universal by default to begin with, as I don't think most people want both an Intel and arm64 mac toolchain to begin with. That base preset used by `build-toolchain` should only build for the host, as it always did until this flag was added last year.

I think Mishal just added it for the CI, to test cross-compiling the toolchain for arm64 too, but unfortunately that is also [the command we prescribe for devs building the toolchain](https://github.com/apple/swift#building), sticking them with an interminable build of both toolchains when they will only want to build for the host.

@johnno1962, try commenting out that flag I linked a couple days ago and only building the toolchain for arm64, without this patch. If that works fine, you should submit a pull for the main Swift repo changing the presets somehow.

Reply:
Thanks @buttaface and @johnno1962 — it's good that it no longer depends on what the host machine is, but it's slightly unfortunate that passing `macosx-arm64` on a M1 machine still results in a universal build whereas passing `macosx-x86_64` doesn't.  However, I think that the fact that `--cross-compile-hosts` is being passed at all is a strong sign that the caller wants a universal toolchain, so it almost doesn't matter what gets passed as a parameter there.  Perhaps the presence of the flag at all should be considered a boolean, i.e. if there's any value at all passed for that parameter, then you get a universal build.  That's the only further change I would suggest here.

Reply:
Thanks @abertelrud, I've pushed a commit that only checks for the presence of cross_compile_hosts which I agree is probably the safest option at this stage. The "else" side of the "if" doesn't quite make sense now but this small section of code can be revisited if there is ever a time when more than two architectures need to be supported.

Reply:
Thanks @johnno1962.  Based on @buttaface's comment below it seems that might have provided poor advice here, in light of PR #3403 that is up to add Android support here.  I will read that in more detail in reply below here.